### PR TITLE
Whd 1 create a booking

### PR DESCRIPTION
**Pull request by JaimySchatteman**

### CODE REVIEW

File: apps/office-service/src/app/app.module.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        } = configService.get('database');
+        return {
+          type: 'postgres',
+          host,
+          port,
+          username,
+          password,
+          database,
+          entities: [Office, Booking],
+          synchronize: true, // TODO turn off synchronization when going into production
```
Comment (darenmalfait) [inline apps/office-service/src/app/app.module.ts:33]: Maybe it's an idea to make the option configurable through en env variable with the default option being false so this never gets into production.

File: apps/office-service/src/booking/dto/create-booking.dto.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { IsDateString, IsOptional, IsString } from 'class-validator';
+
+export class CreateBookingDto {
+  @IsDateString()
+  /*
+   *   TODO Write a custom validator that checks if the date in the
```
Comment (darenmalfait) [inline apps/office-service/src/booking/dto/create-booking.dto.ts:6]: Can we put these TODO's in stories?

File: docker-compose.yml [hunk 3]
```diff
@@ -7,6 +7,9 @@ services:
     image: postgres
     restart: always
     environment:
+      POSTGRES_USER: developer
       POSTGRES_PASSWORD: development
```
Comment (darenmalfait) [inline docker-compose.yml:11]: Are Docker Secrets something that needs to be used for this? @rafael-mindreau

**Review by darenmalfait — CHANGES_REQUESTED** [review]

File: apps/office-service/src/app/app.module.ts (hunk 1 above)
Comment (JaimySchatteman) [reply apps/office-service/src/app/app.module.ts:33]: I'll add to the configuration file

File: apps/office-service/src/booking/dto/create-booking.dto.ts (hunk 2 above)
Comment (JaimySchatteman) [reply apps/office-service/src/booking/dto/create-booking.dto.ts:6]: On it!

File: docker-compose.yml (hunk 3 above)
Comment (RoelGo) [reply docker-compose.yml:11]: The docker compose file is just used for development. No security concerns  here.

**Review by RoelGo — APPROVED** [review]

File: docker-compose.yml (hunk 3 above)
Comment (darenmalfait) [reply docker-compose.yml:11]: this is code going into production. It's not nice to have dev code in production.

Comment (rafael-mindreau) [reply docker-compose.yml:11]: Normally you don't want to push credentials to Git and you want to keep those secrets in `.env` that you gitignore, or keep the secrets in a Vault. However, this case is an exception. This is for a developer to be able to easily run a postgres instance on their own machine. The credentials only make sense for your own machine. So no data can be stolen this way.

An alternative way to keep using Docker would be to supply a `docker run` command in the README that developers can use, but it's way easier to just throw out a `docker-compose up -d` to start developing right away. If your OCD is still triggered, you can put these credentials in `.env` and then you would refer to the environment variable like so:

```yml
environment:
  POSTGRES_USER: ${POSTGRES_USER}
  POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
```

But then again, that's just for OCD's. For me you can do what you want here 😅

Comment (darenmalfait) [reply docker-compose.yml:11]: I think the README option is fine, because then you can also document to why the docker file exists in the first place.

Comment (darenmalfait) [reply docker-compose.yml:11]: Let's go for the environment option. Good practise for when it needs to be done like that.

Comment (darenmalfait) [reply docker-compose.yml:11]: make sure to put some documentation around it in the README aswell.

File: apps/office-service/src/main.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * This is only a minimal backend to get started.
+ */
+
+import { Logger } from '@nestjs/common';
+import { NestFactory } from '@nestjs/core';
+
+import { AppModule } from './app/app.module';
+
+async function bootstrap() {
+  const app = await NestFactory.create(AppModule, { cors: true });
```
Comment (darenmalfait) [inline apps/office-service/src/main.ts:12]: Wat doet cors: true precies? Zet dat alles open?

Comment (darenmalfait) [reply apps/office-service/src/main.ts:12]: Kunnen we een ticket aanmaken om dat achteraf dicht te timmeren?

**Review by darenmalfait — CHANGES_REQUESTED** [review]

File: apps/office-service/src/main.ts (hunk 4 above)
Comment (JaimySchatteman) [reply apps/office-service/src/main.ts:12]: Yes laat dat volledig toe dan!

Comment (JaimySchatteman) [reply apps/office-service/src/main.ts:12]: Ik maakt het ticket aan

**Review by darenmalfait — APPROVED** [review]

**Pull request merged**